### ISSUE TITLE
Séparer les boutons "s'inscrire" et "se connecter"

### ISSUE THREAD

**Issue by overflorian** [vdct/ProjetDuMois#167]
Pour l'instant, lors de la première visite du site, il n'y a qu'un seul bouton "s'inscrire / se connecter" qui renvoit vers https://www.openstreetmap.org/login

Ça renvoit vers un site externe à pdm.fr, sur une page d'inscription, pas de création de compte.
Je pense à notre persona [Mickaël](https://github.com/vdct/ProjetDuMois/issues/66), débutant qui n'a pas encore de compte OSM. Pour lui, ça peut porter à confusion.

Je propose :
1. de spliter les boutons "s'inscrire" et "se connecter" en deux. En renvoyant vers des URLs séparées : https://www.openstreetmap.org/login et https://www.openstreetmap.org/user/new
C'est le même design que sur osm.org
![image](https://user-images.githubusercontent.com/11553211/108096988-dca7ab00-7081-11eb-97ae-aae08ce219ef.png)

2. De créer une popup lorsque l'on clique sur "s'inscrire" pour expliquer pourquoi on va demander la création d'un compte OSM.
Un simple message du type "Ce site utilise votre compte OpenStreetMap pour faire des statistiques et créer le podium des meilleurs contributeurs.
BOUTON "Créer un compte sur OpenStreetMap.org"

Je peux designer un écran avec des flèches explicatives, comme sur cet exemple :
![image](https://user-images.githubusercontent.com/11553211/108099176-93a52600-7084-11eb-9ad0-2bf42c7231c5.png)

3. De splitter de la même manière le bouton de connexion dans la page :
![image](https://user-images.githubusercontent.com/11553211/108097187-1c6e9280-7082-11eb-9294-e69c39a1a80f.png)

D'ailleurs, j'ai une autre idée complémentaire. Je la détaille ici :

PS : [le bouton Facebook connect](https://developers.facebook.com/docs/facebook-login/web?locale=fr_FR#loginbutton) ressemble à ça :
![image](https://user-images.githubusercontent.com/11553211/108098657-ea5e3000-7083-11eb-9bdd-74f1ffd41ce0.png)
Peut-être que je fait fausse route avec ma proposition et que l'on veut conserver un seul bouton "Continuer avec OpenStreetMap" ??

**Comment by overflorian** [comment]
Sur mobile, c'est un peu délicat, les 2 boutons ne rentreraient pas en largeur.
2 solutions :
- on ne change rien pour le mobile
- on renvoit vers la popup intermédiaire pour le mobile, avec en bas de popup, 2 boutons "se connecter" / "créer un compte"
Je penche pour la seconde solution perso.
Question de priorité : ça fait perdre un clic aux geeks mais facilite la vie aux noobs.

**Comment by PanierAvide** [comment]
Si c'est le même fonctionnement sur PC et mobile ça m'arrange :wink: Le principe de popup intermédiaire me paraît choquant, c'est comme ça dans Mapcontrib et ça fait le taf. Je te laisse réfléchir au wording et éléments visuels ?

**Comment by overflorian** [comment]
Yes, je peux faire ça. Du coup @PanierAvide tu pencherai plutôt pour le bouton unique ou pour un split ?
Après réflexion perso je préfère un seul bouton "continuer avec OpenStreetMap".

**Comment by PanierAvide** [comment]
Un bouton unique c'est bien ;-)

--
Adrien


25 févr. 2021 15:03:22 Overflorian <notifications@github.com>:

**Comment by overflorian** [comment]
Tu dis ça parce que c'est plus facile à développer ! Mais en terme d'UX ça fonctionne bien donc tant mieux.

**Comment by overflorian** [comment]
OK le bouton unique est en fait un double bouton (ça serait trop facile) si on suit ce que fait Facebook (cf. https://developers.facebook.com/docs/facebook-login/web/login-button?locale=fr_FR) :
<img width="40%" src="https://user-images.githubusercontent.com/11553211/109174207-ebc6e100-7784-11eb-9476-da809aa437ec.png">
<img width="40%" src="https://user-images.githubusercontent.com/11553211/109174232-f1bcc200-7784-11eb-9322-dd8d13b410ea.png">

Pour le deuxième j'en parle sur une issue séparée : https://github.com/vdct/ProjetDuMois/issues/177

Pour le premier, voici que que j'ai fait :
<img width="40%" src="https://user-images.githubusercontent.com/11553211/109176644-4feaa480-7787-11eb-9caa-0bea364ef2b1.png">

J'ai repris le logo brut produit par osm-fr mais il n'est pas adapté pour une présentation "logo" en petit. Je n'ai pas trouvé mieux pour l'instant ... je vais chercher car je ne suis pas fan du résultat.

**Comment by overflorian** [comment]
OK j'ai simplifié, ça ne suffisait pas, donc j'ai encore plus simplifié ...

<img width="40%" src="https://user-images.githubusercontent.com/11553211/109187643-621e1000-7792-11eb-8b69-3305be5db3c8.png">

Banco pour la 5 ?
Le svg est ici [openstreepmap-logo-blanc-mini.zip](https://github.com/vdct/ProjetDuMois/files/6044561/openstreepmap-logo-blanc-mini.zip)

**Comment by overflorian** [comment]
OK pour l'interstitiel : simple, basique :
<img width="40%" src="https://user-images.githubusercontent.com/11553211/109293065-91cd2680-782b-11eb-81a6-7f9ca48ca312.png">

avec sa variante plus marrante :
<img width="40%" src="https://user-images.githubusercontent.com/11553211/109293896-b7a6fb00-782c-11eb-9b9b-3b46a25974e3.png">


Je propose de bien faire les choses en s'inspirant du material design à fond : https://material.io/components/dialogs#specs
En respectant les tailles :
![image](https://user-images.githubusercontent.com/11553211/109290291-b6bf9a80-7827-11eb-9418-7a2b3832f925.png)

En respectant les couleurs :
![image](https://user-images.githubusercontent.com/11553211/109290696-3d747780-7828-11eb-8dd6-96c86e6fa6cf.png)
Et notamment [notre couleur primaire](https://github.com/vdct/ProjetDuMois/issues/83) : #007bff

--
Wording : 
Ce site a besoin de se connecter
à votre compte OpenStreetMap
pour créer des données et faire
des statistiques.
Promis, on sera sages.

Icônes utilisées : 
[logos pour écran de connexion.zip](https://github.com/vdct/ProjetDuMois/files/6049237/logos.pour.ecran.de.connexion.zip)

**Comment by gendy54** [comment]
Fan de la variante marrante.

Modif cosmétique: "Ce site a besoin de se connecter à votre compte OpenStreetMap pour créer **les** données et faire **vos** statistiques (_de contribution_)"
But: bien différencier les stats du projet, des stats via les cookies que tous les sites ramassent sur nous.
